packages/dara-core/dara/core/interactivity/data_variable.py

```diff
@@ -214,10 +214,11 @@ async def get_value(
         if entry.data is not None:
             filtered_data, count = apply_filters(entry.data, coerce_to_filter_query(filters), pagination)
             if format_for_display and filtered_data is not None:
+                filtered_data = filtered_data.copy()
                 for col in filtered_data.columns:
                     if filtered_data[col].dtype == 'object':
                         # We need to convert all values to string to avoid issues with displaying data in the Table component, for example when displaying datetime and number objects in the same column
-                        filtered_data[col] = filtered_data[col].apply(str)
+                        filtered_data.loc[:, col] = filtered_data[col].apply(str)
```

Comment:
Will this still work for `MultiIndex` rows/`MultiIndex` columns?

Reply:
I have tested with MultiIndex and as far as I can tell seems to work without any issues